### PR TITLE
add labels and annotations to ServingRuntimePodSpec

### PR DESCRIPTION
**Pull request by lizzzcai**
**What this PR does / why we need it**:
* Add `labels` and `annotations` to `ServingRuntimePodSpec`, to pass custom labels and annotations from `servingruntime` to the `pod`
* Added unit test to cover merging labels and annotations

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

https://github.com/kserve/modelmesh-serving/issues/224

**Type of changes**
Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

**Feature/Issue validation/testing**:

Unit test included

**Special notes for your reviewer**:

1. changes needed in modelmesh controller to make it work in modelmesh usecase.

**Checklist**:

- [x] Have you added unit/e2e tests that prove your fix is effective or that this feature works?
- [x] Has code been commented, particularly in hard-to-understand areas?
- [ ] Have you made corresponding changes to the documentation?

**Release note**:

```release-note
Add labels and annotations to ServingRuntimePodSpec, make it easy to pass custom labels and annotations in ModelMesh ServingRuntime.
```
Signed-off-by: Lize Cai <lize.cai@sap.com>

### CODE REVIEW

File: pkg/controller/v1beta1/inferenceservice/utils/utils.go [hunk 1]
```diff
@@ -21,11 +21,12 @@ import (
 	"context"
 	"encoding/json"
 	"html/template"
-	"k8s.io/apimachinery/pkg/util/strategicpatch"
 	"regexp"
 	"sort"
 	"strings"
 
+	"k8s.io/apimachinery/pkg/util/strategicpatch"
```
Comment (yuzisun) [inline pkg/controller/v1beta1/inferenceservice/utils/utils.go]: Looks like this is not used.

File: pkg/controller/v1beta1/inferenceservice/components/predictor.go [hunk 2]
```diff
@@ -224,14 +232,15 @@ func (p *Predictor) Reconcile(isvc *v1beta1.InferenceService) (ctrl.Result, erro
 		annotations[constants.StorageInitializerSourceUriInternalAnnotationKey] = *sourceURI
 	}
 
+	// Labels and annotations from isvc will overwrite labels and annotations from ServingRuntimePodSpec
```
Comment (yuzisun) [inline pkg/controller/v1beta1/inferenceservice/components/predictor.go:235]: Can we add a test case for this?

File: pkg/controller/v1beta1/inferenceservice/utils/utils.go (hunk 1 above)
Comment (lizzzcai) [reply pkg/controller/v1beta1/inferenceservice/utils/utils.go]: it is used in the [code](https://github.com/kserve/kserve/blob/5bcb4e75d1a4de0e7101ceb69955b7ec4bb85c2c/pkg/controller/v1beta1/inferenceservice/utils/utils.go#L96), changes in this file is due to `go fmt`. I can revert them if needed.

File: pkg/controller/v1beta1/inferenceservice/components/predictor.go (hunk 2 above)
Comment (lizzzcai) [reply pkg/controller/v1beta1/inferenceservice/components/predictor.go:235]: @yuzisun added in the controller_test

File: pkg/controller/v1beta1/inferenceservice/components/predictor.go [hunk 3]
```diff
@@ -63,6 +63,8 @@ func NewPredictor(client client.Client, scheme *runtime.Scheme, inferenceService
 func (p *Predictor) Reconcile(isvc *v1beta1.InferenceService) (ctrl.Result, error) {
 	var container *v1.Container
 	var podSpec v1.PodSpec
+	sRuntimeLabels := map[string]string{}
+	sRuntimeAnnotations := map[string]string{}
```
Comment (njhill) [inline pkg/controller/v1beta1/inferenceservice/components/predictor.go]:
```suggestion
	var sRuntimeLabels map[string]string
	var sRuntimeAnnotations map[string]string
```
No need to create empty maps here

**Review by njhill — COMMENTED** [review]
Thanks @lizzzcai, this looks great! Just one small nit from me.

Once this is in we can make the equivalent update on the modelmesh-serving side.

**Comment by yuzisun** [comment]
Thanks @lizzzcai !! Great work!

/lgtm
/approve

**Pull request merged**